api/ssh.go

```diff
@@ -0,0 +1,53 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type Ssh struct {
+	c *Client
+}
+
+func (c *Client) Ssh() *Ssh {
+	return &Ssh{c: c}
+}
+
+func (c *Ssh) KeyCreate(role string, data map[string]interface{}) (*Secret, error) {
+	r := c.c.NewRequest("PUT", fmt.Sprintf("/v1/ssh/creds/"+role))
```

Comment:
Since you are already using the `Sprintf` function, I would use `%s` instead of concatenation here:

```go
fmt.Sprintf("/v1/ssh/creds/%s", role)
```

Does that make sense?